Alex GH Action is failing on forks

**Describe the bug**
[This PR](https://github.com/GoogleChrome/web.dev/actions/runs/207079195) is failing when Alex tries to run. The error says: `##[error]Resource not accessible by integration`.

From a bit of Googling it seems like maybe we need to change the workflow to use the `pull_request_target` event instead of `pull_request` so things can have the proper permissions.
https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/#improvements-for-public-repository-forks

I'm still a bit unclear on how this works but we can land a PR and see if it fixes it 🤷‍♂️

--- a/.github/workflows/lint-and-test-workflow.yml
+++ b/.github/workflows/lint-and-test-workflow.yml
@@ -1,8 +1,10 @@
 name: Lint and test
 
-# By default, runs when a pull_request's activity type is opened, synchronize,
-# or reopened
-on: pull_request
+# By default, runs when a pull request is opened, synchronized, or reopened.
+# We use the pull_request_target hook to enable proper permissions for
+# pull requests coming from forks.
+# https://github.blog/2020-08-03-github-actions-improvements-for-fork-and-pull-request-workflows/#improvements-for-public-repository-forks
+on: pull_request_target
 
 jobs:
   lint_and_test:

